flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/PointwisePatternTest.java

```diff
@@ -352,28 +254,47 @@ private void testHighToLow(int highDop, int lowDop) throws Exception {
             fail("Job failed with exception: " + e.getMessage());
         }
 
-        ExecutionJobVertex target = eg.getAllVertices().get(v2.getID());
+        return eg.getAllVertices().get(v2.getID());
+    }
 
-        int[] timesUsed = new int[highDop];
+    /** Make sure the descendant logic of building POINTWISE edges follows the initial logic. */
+    private void testConnectionFromLowToHigh(int low, int high, int[] expected) throws Exception {
```

Comment:
I think there is no need to distinguish `highToLow` or `lowToHigh`. It can be:

```
int testConnections(int sourceParallelism, int targetParallelism, int[][] expectedTargetConsumedPartitions);
```

The `expectedTargetConnections` describes the `partitionNumber` of consumed partitions for each target parallel instance.